src/server/services/user/user.repository.ts

```diff
@@ -156,6 +164,12 @@ const userRepository = {
           .select({ count: count() })
           .from(users)
           .leftJoin(profiles, eq(users.id, profiles.userId))
+          .leftJoin(
+            profilesAddresses,
+            eq(profiles.id, profilesAddresses.profileId),
+          )
```

Comment:
In `profilesAddresses` we store an additional field called `isPrimary` which allows us to have multiple addresses stored against the user (prebivalište, boraviše).

This query would work perfectly fine if you had only one address stored against the user, but if there were two addresses you would return a user twice due to the way SQL JOINS works.

You can try and add an `and` statement inside of the query e.g.
```ts
and(eq(profiles.id, profilesAddresses.profileId), eq(profilesAddresses.isPrimary, true))
```
Note: I might have written the field names wrong so double check just to be sure :)

Reply:
I've added it now. I also included the ability to disable city filter by clicking the blank option in the dropdown.